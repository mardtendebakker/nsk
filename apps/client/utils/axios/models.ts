export interface BaseCustomer {
  id?: number,
  name?: string,
  representative?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
}

export interface Customer extends BaseCustomer {
  kvk_nr?: string,
  is_partner?: number,
}

export interface Company {
  id: string,
  name: string,
  acompany?: { id: string, name: string }
}

export interface Order {
  id?: number,
  order_nr?: string
  remarks?: string,
  order_date: string,
  discount?: number,
  transport?: number,
  is_gift?: boolean,
  delivery_instructions?: string
  order_status: { name: string, color: string }
  acompany_aorder_supplier_idToacompany?: Company,
  acompany_aorder_customer_idToacompany?: Company
}

export interface AttributeOption {
  id: number,
  name: string,
  price: number
}

export interface Attribute {
  id?: number,
  name: string,
  type: number,
  options: AttributeOption[]
}

export interface ProductType {
  id?: number,
  name: string,
  attributes: Attribute[]
}

export interface Task {
  name: string,
  description: string,
  status: number
}

export interface Product {
  id?: number,
  hold: number,
  location: string,
  name: string,
  order_date: string,
  order_nr: string,
  price: number,
  purch: number,
  sale: number,
  sku: string,
  sold: number,
  splittable: boolean,
  stock: number,
  tasks: Task[],
  updated_at: string
}

export interface Supplier extends BaseCustomer {
  partner?: string,
}

export interface User {
  id?: string,
}
