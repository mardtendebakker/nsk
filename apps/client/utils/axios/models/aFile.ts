export interface AFile {
  id?: number,
  unique_server_filename?: string,
  original_client_filename?: string,
  discr?: string
}
