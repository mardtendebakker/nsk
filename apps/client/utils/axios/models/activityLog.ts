export interface ActivityLog {
  id: number;
  username?: string;
  method?: string;
  route?: string;
  params?: string;
  body?: string;
  query?: string;
  before?: string;
  model?: string;
  action?: string;
  bulk: boolean;
  createdAt: string;
  updated_at: string;
}
