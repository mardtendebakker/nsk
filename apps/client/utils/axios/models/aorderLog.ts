export interface OrderStatus {
  id: number;
  name: string;
  color: string;
}

export interface AorderLog {
  id: number;
  username: string;
  previous_status_id: number;
  status_id: number;
  previous_status: OrderStatus;
  status: OrderStatus;
  created_at: string;
  updated_at: string;
}
