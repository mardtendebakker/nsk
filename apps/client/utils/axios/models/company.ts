export interface CompanyListItem {
  id:number,
  name:string,
  kvk_nr:string,
  is_partner:boolean,
  is_supplier:boolean,
  is_customer:boolean,
  contactsCount:number
}

export interface Company {
  id:number,
  name:string,
  kvk_nr:string,
  is_partner:boolean,
  is_supplier:boolean,
  is_customer:boolean,
  partner_id:number,
  tax_code:number,
}
