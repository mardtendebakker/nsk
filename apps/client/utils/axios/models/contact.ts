export interface ContactListItem {
  id: number,
  name:string,
  company_name:string,
  email:string,
  partner:{ id: number, name: string },
  is_partner:boolean,
  is_supplier:boolean,
  is_customer:boolean,
  ordersCount:number
}

export interface Contact {
  id:number,
  name:string,
  company_id: number,
  company_kvk_nr:string,
  email:string,
  phone:string,
  phone2:string,
  street:string,
  street_extra:string,
  city:string,
  country:string,
  state:string,
  zip:string,
  street2:string,
  street_extra2:string,
  city2:string,
  country2:string,
  discr:string,
  state2:string,
  zip2:string,
  is_partner:boolean,
  is_customer:boolean,
  is_supplier:boolean,
  partner_id:number,
}
