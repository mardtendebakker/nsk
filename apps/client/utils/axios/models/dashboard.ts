export interface DashboardTotal {
  totalSpent?: number;
  totalEarned?: number;
  totalOrders?: number;
  totalSuppliers?: number;
  totalCustomers?: number;
}
