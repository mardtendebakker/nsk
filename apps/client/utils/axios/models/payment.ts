import { ModulePayment } from './modulePayment';

export interface PaymentListItem {
  id: number;
  method?: string;
  transactionId: string;
  subscriptionId?: string;
  amount: number;
  status: string;
  createdAt: string;
  updatedAt: string;
  modules: ModulePayment[]
}
