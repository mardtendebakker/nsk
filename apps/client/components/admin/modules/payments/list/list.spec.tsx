import React from 'react';
import {
  render, screen, waitFor,
} from '@testing-library/react';
import List from './list';
import { PaymentListItem } from '../../../../../utils/axios/models/payment';

jest.mock('../../../../../hooks/useTranslation', () => ({
  __esModule: true,
  default: () => ({
    trans: (key) => key,
  }),
}));

const mockPayment: PaymentListItem = {
  id: 1,
  method: 'credit_card',
  transactionId: 'txn_123',
  amount: 1000,
  status: 'paid',
  createdAt: '2024-01-01T00:00:00Z',
  updatedAt: '2024-01-01T00:00:00Z',
  subscriptionId: null,
  modules: [
    {
      id: 1,
      name: 'Test Module',
      price: 500,
      activeAt: '2024-01-01T00:00:00Z',
      expiresAt: '2024-12-31T00:00:00Z',
    },
  ],
};

const defaultProps = {
  payments: [mockPayment],
  disabled: false,
  count: 1,
  page: 1,
  rowsPerPage: 10,
  onPageChange: jest.fn(),
  onRowsPerPageChange: jest.fn(),
  onSubscribe: jest.fn(),
};

describe('List Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<List {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the payment list with correct data', () => {
    render(<List {...defaultProps} />);

    expect(screen.getByText('method')).toBeInTheDocument();
    expect(screen.getByText('transactionId')).toBeInTheDocument();
    expect(screen.getByText('amount')).toBeInTheDocument();
    expect(screen.getByText('status')).toBeInTheDocument();
    expect(screen.getByText('credit_card')).toBeInTheDocument();
    expect(screen.getByText('txn_123')).toBeInTheDocument();
  });

  it('shows different chip variants based on payment status', () => {
    const payments = [
      { ...mockPayment, id: 1, status: 'paid' },
      { ...mockPayment, id: 2, status: 'pending' },
      { ...mockPayment, id: 3, status: 'refunded' },
    ];

    render(<List {...defaultProps} payments={payments} />);

    const chips = screen.getAllByText(/(paid|pending|refunded)/);
    expect(chips).toHaveLength(3);
  });

  it('disables interactions when disabled prop is true', () => {
    render(<List {...defaultProps} disabled />);

    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toBeDisabled();
  });

  it('handles empty payment list', () => {
    render(<List {...defaultProps} payments={[]} />);

    expect(screen.getByText('method')).toBeInTheDocument();
    expect(screen.getByText('transactionId')).toBeInTheDocument();
    expect(screen.queryByText('credit_card')).not.toBeInTheDocument();
  });

  it('formats prices correctly', () => {
    render(<List {...defaultProps} />);
    waitFor(() => {
      expect(screen.getByText('€ 1.000,00')).toBeInTheDocument();
    });
  });
});
