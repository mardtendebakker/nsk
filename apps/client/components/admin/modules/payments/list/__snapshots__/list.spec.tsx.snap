// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component matches snapshot 1`] = `
<DocumentFragment>
  <table
    class="MuiTable-root css-rqglhn-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
          scope="col"
        >
          method
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
          scope="col"
        >
          transactionId
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
          scope="col"
        >
          amount
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
          scope="col"
        >
          status
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
          scope="col"
        >
          createdAt
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
          scope="col"
        >
          updatedAt
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
          scope="col"
        >
          recurrentPayment
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-hover css-rj1dqh-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
        >
          credit_card
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
        >
          txn_123
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
        >
          € 1.000,00
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
        >
          <div
            class="MuiBox-root css-1nzqqgc"
          >
            paid
          </div>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
        >
          2024/01/01
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
        >
          2024/01/01
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
        >
          <div
            class="MuiBox-root css-70qvj9"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-1w6ohhk-MuiTableCell-root"
          colspan="7"
        />
      </tr>
    </tbody>
  </table>
  <div
    class="MuiTablePagination-root css-1a5aq4b-MuiTablePagination-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
    >
      <div
        class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
      />
      <p
        class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
        id=":r2:"
      >
        Rows per page:
      </p>
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby=":r2: :r1:"
          class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
          id=":r1:"
          role="button"
          tabindex="0"
        >
          10
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="10"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <p
        class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
      >
        1–1 of 1
      </p>
      <div
        class="MuiTablePagination-actions"
      >
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          disabled=""
          tabindex="-1"
          title="Go to previous page"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardArrowLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
            />
          </svg>
        </button>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          disabled=""
          tabindex="-1"
          title="Go to next page"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
